### PR TITLE
Added support for customizing the way `Display` works on `DeriveActiveEnum`

### PR DESCRIPTION
**Pull request by asdfcube**
## PR Info

- Closes https://github.com/SeaQL/sea-orm/issues/1255

## New Features

- [x] added `display`, `display_value`, `display_label` attributes to the `#[sea_orm()]` container and variant attribute.
- [x] `DeriveActiveEnum` only implements `std::fmt::Display` trait automatically if one of the attributes are present.
- [x] made users able to customise the display name by using the variant attribute `display = "string literal"`.

## BREAKING CHANGES

- [x] `#[derive(DeriveActiveEnum)]` no longer derives `std::fmt::Display` automatically if `display` or `display_value` or `display_label` are not present in the container attributes.
- [x] `display` defaults to displaying label instead of value now

### CODE REVIEW

File: src/entity/active_enum.rs [hunk 1]
```diff
@@ -275,6 +275,64 @@ mod tests {
...
+        #[sea_orm(rs_type = "i32", db_type = "Integer")]
+        enum DefaultDisplay {
+            One = 1,
+            Two = 2,
+            Three = 3,
+        }
+
+        assert_eq!(format!("{}", DefaultDisplay::One), "1");
+        assert_eq!(format!("{}", DefaultDisplay::Two), "2");
+        assert_eq!(format!("{}", DefaultDisplay::Three), "3");
```
Comment (tyt2y3) [inline src/entity/active_enum.rs:291]: As per our discussion, I think we should remove the default Display impl, resulting in a breaking change.

File: src/entity/active_enum.rs [hunk 2]
```diff
@@ -275,6 +275,64 @@ mod tests {
...
+
+        // All `label` -> All print label
+        #[derive(PartialEq, Eq, EnumIter, DeriveActiveEnum)]
+        #[sea_orm(rs_type = "i32", db_type = "Integer", display = "label")]
```
Comment (tyt2y3) [inline src/entity/active_enum.rs:295]: "label" can be the default. May be we can make `display` an alias of `display = "label"`.

File: src/entity/active_enum.rs [hunk 3]
```diff
@@ -275,6 +275,64 @@ mod tests {
...
+
+        assert_eq!(format!("{}", LabelDisplay::One), "'One'");
+        assert_eq!(format!("{}", LabelDisplay::Two), "'Two'");
+        assert_eq!(format!("{}", LabelDisplay::Three), "'Three'");
+
+        // Default with override
+        #[derive(PartialEq, Eq, EnumIter, DeriveActiveEnum)]
+        #[sea_orm(rs_type = "i32", db_type = "Integer")]
+        enum DefaultDisplayOverride {
+            #[sea_orm(display = "label")]
```
Comment (tyt2y3) [inline src/entity/active_enum.rs:310]: Actually, I think this override should be the thing the user wants?

So `display = "hello"` should result in 

```rust
assert_eq!(format!("{}", DefaultDisplayOverride::One), "'hello'");
```

File: src/entity/active_enum.rs [hunk 4]
```diff
@@ -275,6 +275,64 @@ mod tests {
...
+        // All `label` -> All print label
+        #[derive(PartialEq, Eq, EnumIter, DeriveActiveEnum)]
+        #[sea_orm(rs_type = "i32", db_type = "Integer", display = "label")]
+        enum LabelDisplay {
+            One = 1,
+            Two = 2,
+            Three = 3,
+        }
+
+        assert_eq!(format!("{}", LabelDisplay::One), "'One'");
```
Comment (tyt2y3) [inline src/entity/active_enum.rs:302]: Umm... I don't have a clear sense, but seems like the `'` is unnecessary in this context? Like the expected should be:

```rust
assert_eq!(format!("{}", LabelDisplay::One), "One");
```

File: src/entity/active_enum.rs (hunk 1 above)
Comment (asdfcube) [reply src/entity/active_enum.rs]: done in 656c777e8c7d33d2eb4f316be3b7627f07d5c72f

File: src/entity/active_enum.rs (hunk 2 above)
Comment (asdfcube) [reply src/entity/active_enum.rs]: as discussed, we want `display = label | value | "string literal"` but i cannot get it working, so instead went to `display = "string literal" | display_value | display_label`.

File: src/entity/active_enum.rs (hunk 3 above)
Comment (asdfcube) [reply src/entity/active_enum.rs]: that relied on the implementation of `Value::to_string`, so...

File: src/entity/active_enum.rs (hunk 4 above)
Comment (asdfcube) [reply src/entity/active_enum.rs]: that relied on the implementation of `Value::to_string`, so...

File: src/entity/active_enum.rs (hunk 3 above)
Comment (asdfcube) [reply src/entity/active_enum.rs]: done in 656c777e8c7d33d2eb4f316be3b7627f07d5c72f

File: src/entity/active_enum.rs (hunk 1 above)
Comment (asdfcube) [reply src/entity/active_enum.rs]: done in 93cdf20277d1cc4294bf5a8301b5daff6a2639f4

File: src/entity/active_enum.rs (hunk 4 above)
Comment (asdfcube) [reply src/entity/active_enum.rs]: as discussed, we will keep the behaviour

**Comment by tyt2y3** [comment]
@billy1624 @darkmmon please put this on the task list

**Comment by billy1624** [comment]
Sure. I want him to touch on tangible API before entering the magical proc_macro world

**Pull request closed without merging**